### ISSUE TITLE
HalogenHooks recipes are broken on TryPureScript

### ISSUE THREAD

**Issue by ptrfrncsmrph** [JordanMartinez/purescript-cookbook#279]
**Recipe Name**

```
*HalogenHooks
```

**Reason for Breakage / Steps to Fix**

I think all the `*HalogenHooks` are currently broken on TryPureScript because of the mismatch of Halogen versions. Seems like a lot of them could be fixed by removing `HH.HTML` as the first argument to `Component` (which now just takes four type parameters: https://pursuit.purescript.org/packages/purescript-halogen/6.1.2/docs/Halogen#t:Component), and removing `Just` from some of the event handlers (`Halogen.HTML.Event.onClick` for example now takes `MouseEvent -> i` not `MouseEvent -> Maybe i`: https://pursuit.purescript.org/packages/purescript-halogen/6.1.2/docs/Halogen.HTML.Events#v:onClick)

**Comment by ptrfrncsmrph** [comment]
I'm not well-versed in Halogen, but I think I can at least fix the trivially-fixable ones and mark the rest as broken. I'll plan on having a PR sometime this week.

**Comment by JordanMartinez** [comment]
Yeah, one of the breaking changes in Halogen was dropping the `surface` type parameter, so that's what's causing a lot of the breakage here.

**Comment by thomashoneyman** [comment]
Those two changes should fix everything except for any recipes that rely on event sources, as those have also changed in Halogen 6 (to become simpler!).

**Comment by ptrfrncsmrph** [comment]
I've started down the road of bumping PureScript to 0.14.3 and the package set to latest: https://github.com/JordanMartinez/purescript-cookbook/pull/280. I'm not sure if there's a better alternative to resolving this, but so far the breakages haven't been _too_ bad, and there might be trivial fixes for the ones I've marked as `broken`.

I can probably get to the rest sometime this week, if this would be a welcome change. If there's an alternative that's more minimal I could look into that instead.
